### PR TITLE
SDI-104: 💄 Only show age if under 18

### PR DESCRIPTION
**Pull request by petergphillips**
New section separating out children from adults
Screenshot from 2022-03-09 21-13-17

### CODE REVIEW

File: backend/controllers/prisonerProfile/prisonerVisits.ts [hunk 1]
```diff
@@ -13,9 +13,9 @@ const sortByListSequenceThenDescription = (left, right): number => {
   return listSeqSort !== 0 ? listSeqSort : compareStrings(left.description, right.description)
 }
 
-const sortByLeadThenAgeThenLastNameFirstName = (left, right): number => {
-  if (right.leadVisitor) return 1
-  if (left.leadVisitor) return -1
+const sortByLeadIf18OrOverThenAgeThenLastNameFirstName = (left, right): number => {
+  if (right.leadVisitor && right.age >= 18) return 1
```
Comment (petergphillips) [inline backend/controllers/prisonerProfile/prisonerVisits.ts:17]: according to policy shouldn't get any lead visitors under 18, but need to ensure they appear sorted in the children section just in cased

File: backend/controllers/prisonerProfile/prisonerVisits.ts [hunk 2]
```diff
@@ -118,20 +131,38 @@ export default ({ prisonApi, pageSize = 20 }) =>
               const start = moment(startTime, DATE_TIME_FORMAT_SPEC)
               const end = moment(endTime, DATE_TIME_FORMAT_SPEC)
 
-              return {
-                isFirst: i === 0,
-                isLast: i + 1 === arr.length,
+              const ageAsText = calculateChildAgeAsText(visitor.age)
+
+              if (i === firstChildIndex) {
+                arr.push({
```
Comment (petergphillips) [inline backend/controllers/prisonerProfile/prisonerVisits.ts:137]: add extra row with `children:` heading

File: integration-tests/integration/prisonerProfile/prisonerVisits.spec.js [hunk 3]
```diff
@@ -129,9 +129,8 @@ context('Prisoner visits', () => {
             expect($tableCells.get(2)).to.contain.text('Social')
             expect($tableCells.get(3)).to.contain.text('Not entered')
             expect($tableCells.get(4)).to.contain.text('Dythispal Alfres')
-            expect($tableCells.get(5)).to.contain.text('37')
```
Comment (petergphillips) [inline integration-tests/integration/prisonerProfile/prisonerVisits.spec.js:132]: age no longer displayed in separate column

File: views/prisonerProfile/prisonerVisits/prisonerVisits.njk [hunk 4]
```diff
@@ -43,12 +43,8 @@
...
     },
     {
-      text: visit.name,
-      classes: 'govuk-table__cell--last' if visit.isLast
-    },
-    {
-      text: visit.age,
-      classes: 'govuk-table__cell--last' if visit.isLast 
+      text: visit.nameWithChildAge,
+      classes: 'govuk-table__cell--last' if visit.isLast or visit.lastAdult
```
Comment (petergphillips) [inline views/prisonerProfile/prisonerVisits/prisonerVisits.njk:47]: extra bar between adults and children

File: backend/controllers/prisonerProfile/prisonerVisits.ts (hunk 1 above)
Comment (andrewrlee) [reply backend/controllers/prisonerProfile/prisonerVisits.ts:17]: I remember Nev saying that in practice children often appear as lead visitors

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**